lib/core/src/Cardano/Wallet/Primitive/Fee.hs

```diff
@@ -357,7 +357,7 @@ coalesceDust threshold coins
         []
     | otherwise =
         let
-            filtered = L.filter (> threshold) coins
+            filtered = L.filter (\x -> x >= threshold && x > Coin 0) coins
```

Comment:
Dusty coins are anything that is **below** the threshold. So we remove dusty coins here (and anything that's `0`) and then split the dust value across the good coins.